src/main/java/io/cdap/plugin/http/HTTPArgumentSetter.java

```diff
@@ -72,8 +72,8 @@ public void run(ActionContext context) throws Exception {
         int responseCode = response.getResponseCode();
 
         LOG.debug("Request to {} resulted in response code {}.", conf.getUrl(), responseCode);
-        if (responseCode / 100 != 2) {
-          throw new IllegalStateException(String.format("Received non-ok response code %d. Response message = %s",
+        if (responseCode != 200) {
```

Comment:
Is there a possibility this response code could be some other success code like 201?

Reply:
I think so, but from the ticket it sounds like we only want to consider 200 (since we were previously allowing all 2xx codes)

Reply:
I talked to Sree about this and the issue is that when its a non-200 response, the macro doesnt get resolved and it fails in the source instead of failing in the action. Now, it will handle response only for 200 else will fail and emit the response code as well.